demo/custom_data_types/arasaka_cyber_corp/PerformLinearLeastSquaresFit.1.0.dsdl

```diff
@@ -3,11 +3,11 @@
 
 PointXY.1.0[<64] points
 
-@extent 1024 * 8
+@extent 16 * 2 * 64 * 4
```

Comment:
I don't understand the purpose of this change. The original statement was correct -- the extent is set (arbitrarily) to 1 KiB. The size is not dependent on the fields contained in the definition, it is rather a top-level design requirement for the data type.

Reply:
> Graphviz comment

```
digraph G {

    subgraph cluster {
        label = "42:org:opencyphal.pycyphal.demo.demo_app";
        node [shape=box]

        subgraph cluster_1 {
            label = "heartbeat_publisher";
            heartbeat_publisher_node[label="uavcan.node.Hearbeat.1.0", shape=trapezium, style=filled]
            uavcan_node_heartbeat[style=filled]
        }
        heartbeat_publisher_node -> uavcan_node_heartbeat
        

        subgraph cluster_2 {
            label = "temperature_setpoint";
            uavcan_si_sample_temperature_scalar_2345[style=filled]
            temperature_setpoint_node[label="uavcan.si.unit.temperature.Scalar_1", shape=invtrapezium, style=filled]
        }
        uavcan_si_sample_temperature_scalar_2345 -> temperature_setpoint_node

        subgraph cluster_3 {
            label = "temperature_measurement";
            uavcan_si_sample_temperature_scalar_2346[style=filled]
            temperature_measurement_node[label="uavcan.si.sample.temperature.Scalar_1", shape=invtrapezium, style=filled]
        }
        uavcan_si_sample_temperature_scalar_2346 -> temperature_measurement_node
        

        subgraph cluster_4 {
            label = "heater_voltage";
            heater_voltage_node[label="uavcan.si.unit.voltage.Scalar_1", shape=trapezium, style=filled]
            uavcan_si_unit_voltage_Scalar[label="2347:uavcan.si.unit.voltage.Scalar", style=filled]
        }
        heater_voltage_node -> uavcan_si_unit_voltage_Scalar

        subgraph cluster_5 {
            label = "least_squares";
            least_squares_service[label="sirius_cyber_corp.PerformLinearLeastSquaresFit_1", shape=hexagon, style=filled]
            sirius_cyber_corp_PerformLinearLeastSquaresFit_1_Request_123[label="123:sirius_cyber_corp.PerformLinearLeastSquaresFit_1.Request", style=filled]
            sirius_cyber_corp_PerformLinearLeastSquaresFit_1_Response_123[label="123:sirius_cyber_corp.PerformLinearLeastSquaresFit_1.Response", style=filled]
        }
        sirius_cyber_corp_PerformLinearLeastSquaresFit_1_Request_123 -> least_squares_service
        least_squares_service -> sirius_cyber_corp_PerformLinearLeastSquaresFit_1_Response_123
        

    }
    
}
```

How to enforce the order of the subgraphs?

Reply:
> The size is not dependent on the fields contained in the definition, it is rather a top-level design requirement for the data type.

So, just to make sure I understand: 

_It can't it set to less than the total size of fields contained in the definition?_

From [Specification](https://opencyphal.org/specification/Cyphal_Specification.pdf):

> Extent — the minimum amount of memory, in bits, that shall be allocated to store the serialized representa-
tion of the type. The extent of any type is always greater than or equal the maximal value of its bit length set. It
is always a multiple of 8.

So chosen value should always be:

- greater/equal to 
  - if equal 
       - no fields can be added later (-> might as well use `sealed` in that case I guess?)
- multiple of 8

_It's not really arbitrary and does depend on the underlying total size?_

Reply:
Yes, your understanding is correct. The extent value used here was chosen rather arbitrarily. You can ask the DSDL front-end to emit the actual size of a type by placing a statement like this after the last field:

```python
@print _offset_.max / 8  # Divide by 8 to obtain bytes (DSDL is bit-oriented)
```

`PointXY` is 4 bytes large and is sealed; 63 (sic! note `<` not `<=`) instances of `PointXY` take 252 bytes, plus one byte for the array length prefix, resulting in 253 bytes total.

You can also do this to obtain the list of all possible lengths at a given offset:

```python
@print _offset_ / 8
```

Be sure to feed `-v` to Nunavut to see the print outputs (actually it shouldn't require verbosity above the default value, perhaps this should be fixed in Nunavut):

```sh
nnvg -v --target-language c sirius_cyber_corp
```

Reply:
At some point I will need to read the entire Specification. 👍